File: backend/shared/uidl/src/main/java/io/mateu/uidl/data/FormLayout.java
package io.mateu.uidl.data;

import io.mateu.uidl.fluent.Component;
import java.util.List;
import lombok.Builder;

@Builder
public record FormLayout(
    String id,
    List<Component> content,
    boolean autoResponsive,
    boolean labelsAside,
    int maxColumns,
    String columnWidth,
    boolean expandColumns,
    boolean expandFields,
    Object responsiveSteps,
    String itemLabelWidth,
    String columnSpacing,
    String itemRowSpacing,
    String itemLabelSpacing,
    String style,
    String cssClasses)
    implements Component {}

File: frontend/web/vaadin/src/mateu/shared/apiClients/dtos/componentmetadata/FormLayout.ts
import ComponentMetadata from "@mateu/shared/apiClients/dtos/ComponentMetadata";

export interface FormLayoutItem extends ComponentMetadata {

}


export interface FormLayoutRow extends ComponentMetadata {

}

export default interface FormLayout extends ComponentMetadata {

    autoResponsive: boolean
    maxColumns: number | undefined
    labelsAside: boolean
    columnWidth: string
    expandColumns: boolean
    expandFields: boolean

    responsiveSteps: any
    itemLabelWidth: string | undefined

    columnSpacing: string | undefined
    itemRowSpacing: string | undefined
    itemLabelSpacing: string | undefined

}
